manim/mobject/svg/svg_mobject.py

```diff
@@ -124,28 +125,28 @@ def _ensure_valid_file(self):
         if self.file_name is None:
             raise Exception("Must specify file for SVGMobject")
 
-        if os.path.exists(self.file_name):
+        if Path(self.file_name).exists():
             self.file_path = self.file_name
```

Comment:
Hi @jbalpert , thanks for this pr! I'll have a look on this in detail soon.
One idea quick idea already: Can you have a look if there is a good solution that the SVGMobject can also accept pathlib objects, e.g. like this?
```py
class Example(Scene):
    def construct(self):
        path = Path.cwd() / "wind.svg"
        em = SVGMobject(path, color=BLACK) # <- not yet working
        #em = SVGMobject(str(path), color=BLACK) # <- current solution
        self.add(em)
  ```

Reply:
Will do! 

Sidenote: Do you have any insights as to why this is failing the pipeline right now?

Reply:
Some paths seem to be broken in Text and MarkupText, but I did not go into it in detail:
```
FAILED manim/mobject/text/text_mobject.py::manim.mobject.text.text_mobject.MarkupText
FAILED manim/mobject/text/text_mobject.py::manim.mobject.text.text_mobject.Text
============= 2 failed, 44 passed, 3 skipped, 46 warnings in 4.85s =============
```

```
1085         >>> MarkupText('The horse does not eat cucumber salad.')
UNEXPECTED EXCEPTION: FileNotFoundError(2, 'No such file or directory')
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/doctest.py", line 1[33](https://github.com/ManimCommunity/manim/runs/6088268786?check_suite_focus=true#step:20:33)7, in __run
    compileflags, 1), test.globs)
  File "<doctest manim.mobject.text.text_mobject.MarkupText[0]>", line 1, in <module>
  File "/home/runner/work/manim/manim/manim/mobject/text/text_mobject.py", line 1146, in __init__
    file_name = self._text2svg(color)
  File "/home/runner/work/manim/manim/manim/mobject/text/text_mobject.py", line 1249, in _text2svg
    Path(dir_name).mkdir()
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/pathlib.py", line 1273, in mkdir
    self._accessor.mkdir(self, mode)
FileNotFoundError: [Errno 2] No such file or directory: 'media/texts'
```

Reply:
My theory: `mkdir(parents=True)` has to be added somewhere.

Reply:
That's a good suggestion! Also, we modified that SVGMobject to take in either a path or a string. Within the constructor, we convert the path or string into a string.

Reply:
Wooo! I fixed the error! I'm guessing it related either .mkdir(parents=True) or .exists() which I changed to .is_dir()

Reply:
That is a great addition to the library, thank you so much for your efforts! 
Some suggestions from my side: 
* The Github CI is not the fastest, so you can also run the tests locally with `pytest`.
* It's often better to make multiple small pull requests  then one large one (I think yours is medium size though ;))
Reason: It's then easier to trace where errors in the test come from and also easier to debug. Another good side effect: There is a lower barrier and less time required to review small pull requests than large pull requests, so small pull requests might get reviewed sooner ;)
* If you want to chat with us in real time, come visit us at https://manim.community/discord

Reply:
> Is it better if each time I add a new change and locally test it I create a new pull request rather than committing to a previous pull request?

When you work on multiple change, it's always good to have multiple pull requests.
E.g. recently I worked on new thumbnails for scikit-image, and for each new thumbnail I created a separate pull request:
https://github.com/scikit-image/scikit-image/pull/6298
https://github.com/scikit-image/scikit-image/pull/6300
https://github.com/scikit-image/scikit-image/pull/6301

But for one feature that you work on, you just keep on adding further commits to your pull request like you do it now, until it's ready to be merged into the library.

Regarding running all tests vs. specific test, you can find a reference here.: https://docs.pytest.org/en/6.2.x/usage.html#specifying-tests-selecting-tests

Reply:
Wow thank you so much! This has been my friend and my first contribution to an opensource project and this has been such an awesome community and great experience!

Reply:
You are very welcome!
We are happy about everybody who joins our community and the mission to create a good tool for science and math communication. Mostly we are talking on Discord https://manim.community/discord, so you are more than welcome to join our discussions about manim and other stuff. 
One thing regarding this pr:
I've noted that you are both altering the library code AND the test.
This should not happen in one pull request, because  the tests that are run by the Github CI are the new tests.
But it might happen that the old tests will show an error with the changed library code, which the new tests won't show anymore, so it's like bypassing the testing system, and this should be avoided.

Reply:
Ohh sounds good! I think when we updated the OS.path instances, we naively searched throughout the entire repo without thinking of much of it. We just issued a new pull request #2687 that details the changes to just the **library code**. Thanks again! We joined the discord yesterday!

Reply:
> We joined the discord yesterday

Amazing, see you around there!
 
> We just issued a new pull request https://github.com/ManimCommunity/manim/pull/2687 that details the changes to just the library code.

Great!
Now there is the small problem, that the changes to the library code is present in both the pull requests.
I therefore recommend that in this pr here you revert the changes in all library code files (there are IDE tools for that, e.g. in Pycharm Or VSCode), and only keep the changes that were done in the test code.
Then, the PRs can be named like this:
#2685 -> Migrate from os.path to pathlib in Testing Scripts
#2687 -> Migrate from os.path to pathlib in SVGMobject and other locations

Reply:
I just resolved that issue so now #2685 is strictly pathlib changes for testing scripts while #2687 is for the changes in library code!